src/main/java/spotify/factories/AbstractPlayableObjectDeserializer.java

```diff
@@ -7,13 +7,46 @@
 
 import java.lang.reflect.Type;
 
+/**
+ * This is a AbstractPlayableObjectDeserializer class to implement JsonDeserializer.
+ *
+ * @see spotify.models.generic.AbstractPlayableObject
+ * @author Jiankai Zheng (jk.zheng@hotmail.com)
+ * @since // check with Jiankai
+ */
+
 class AbstractPlayableObjectDeserializer implements JsonDeserializer<AbstractPlayableObject> {
+
+    /**
+     * Custom deserialize method to deserialize json, type and jsonDeserializationContext
+     * @param json represents a json element
+     * @param type represents various types
+     * @param jsonDeserializationContext
```

Comment:
There is no description

Reply:
Would this be better?

- @param json represents the json element to deserialize
- @param type the type to convert the data into
- @param jsonDeserializationContext the context in which are the elements will be deserialized

Reply:
- no, the first param should just say 'the json to be deserialized'
- second param is fine
- third param, context for deserialization that is passed to a custom deserializer during invocation of its JsonDeserializer.deserialize method

Reply:
Thanks. I will update them.